packages/type-safe-api/docs/developer_guides/type-safe-api/websocket_lambda_handlers.md

```diff
@@ -0,0 +1,310 @@
+    !!!warning
+
+        Python is not yet supported.
+
+When you use a handler router, you must specify the same lambda function for every integration in your `Api` CDK construct. To save typing, you can use the `Operations.all` method from your generated runtime package:
+
+=== "TS"
+
+    ```ts
+    import { Operations } from "myapi-typescript-runtime";
+    import { Api } from "myapi-typescript-infra";
+    import { WebSocketIamAuthorizer } from "aws-cdk-lib/aws-apigatewayv2-authorizers";
+    import { WebSocketLambdaIntegration } from "aws-cdk-lib/aws-apigatewayv2-integrations";
+    import { NodejsFunction } from "aws-cdk-lib/aws-lambda";
+
+    new WebSocketApi(this, "Api", {
+      authorizer: new WebSocketIamAuthorizer(),
+      // Use the same integration for every operation.
+      integrations: Operations.all({
+        integration: new WebSocketLambdaIntegration("RouterIntegration", new NodejsFunction(this, "Router")),
+      }),
+    });
+    ```
+
+=== "JAVA"
+
+    !!!warning
+
+        Java is not yet supported.
+
+=== "PYTHON"
+
+    !!!warning
+
+        Python is not yet supported.
+
+## Connect and Disconnect Handlers
+
+Using the `@connectHandler`/`@disconnectHandler` Smithy traits (or the `x-connect-handler`/`x-disconnect-handler` vendor extensions in OpenAPI) will generate lambda handler stubs for managing connections.
+
+The `$connect` handler is called when a new client attempts to connect to the WebSocket API, and is authenticated. You may optionally choose to deny the connection if you would like to perform additional authorization. Note that the connection is not established (or denied) until the lambda function returns, so you cannot use the Server SDK to send messages to the given connection. You can however use the SDK to send messages to other connections.
+
+Often, the `$connect` handler is used to store connection IDs along with metadata such as the authenticated user, and the `$disconnect` handler is used to clean up any state pertaining to a connection.
+
+=== "TS"
+
+    ```ts
+    import {
+      $connectHandler,
+      $ConnectChainedLambdaHandlerFunction,
+      INTERCEPTORS,
+      LoggingInterceptor,
+      $PendingConnection,
+    } from 'myapi-typescript-runtime';
+
+    /**
+     * Type-safe handler for the $connect event, invoked when a new client connects to the websocket
+     */
+    export const $connect: $ConnectChainedLambdaHandlerFunction = async (request) => {
+      LoggingInterceptor.getLogger(request).info('Start $connect');
+
+      // `connectionId` is the ID of the new connection
+      // `sdk` is used to send messages to connected clients
+      // Note that you cannot send messages to the new connection until after this function returns
+      const { connectionId, sdk } = request;
+
+      // TODO: Implement
+
+      // Use the below to allow or deny the incoming connection (when the lambda returns).
+      // The connection is allowed by default.
+      if (Math.random() > 0.5) {
```

Comment:
maybe replace this with something like: `if (!authenticated(request)) ...`

Reply:
Good shout!